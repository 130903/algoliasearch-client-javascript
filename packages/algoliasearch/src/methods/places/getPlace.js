// @flow

import type { ObjectID } from 'algoliasearch';
import type { RequestMethod, RequestOptions } from 'algoliasearch-requester';

export default function getPlace(
  {
    objectID,
    attributesToRetrieve,
  }: {
    objectID: ObjectID,
    attributesToRetrieve: string[],
  },
  {
    requester,
    requestOptions,
  }: {
    requester: RequestMethod,
    requestOptions?: RequestOptions,
  } = {}
) {
  const attributes = attributesToRetrieve && {
    attributes: attributesToRetrieve.join(','),
  };

  return requester({
    method: 'GET',
    path: `/1/indexes/places/${objectID}`,
    qs: { ...attributes },
    requestType: 'read',
    requestOptions,
  });
}
