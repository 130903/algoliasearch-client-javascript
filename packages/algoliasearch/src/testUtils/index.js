// @flow

import type { RequestMethod } from 'algoliasearch-requester';

export const snapshotAll = (requests: RequestMethod[]) =>
  requests.map(req => req.then(sn => expect(sn).toMatchSnapshot()));
