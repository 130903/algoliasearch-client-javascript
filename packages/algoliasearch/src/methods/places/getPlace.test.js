import { snapshotAll, fakeRequester } from '../../testUtils';
import getPlace from './getPlace';

it('regular getPlaces', () => {
  const requests = [
    getPlace({ requester: fakeRequester, objectID: 'some_item' }),
    getPlace({
      requester: fakeRequester,
      objectID: 'some_item',
      options: {
        attributesToRetrieve: ['one', 'name'],
      },
    }),
  ];
  snapshotAll(requests);
});
