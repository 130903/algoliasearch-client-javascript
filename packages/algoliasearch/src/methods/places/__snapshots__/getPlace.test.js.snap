// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`regular getPlaces 1`] = `
Object {
  "method": "GET",
  "path": "/1/indexes/places/some_item",
  "qs": Object {},
  "requestOptions": undefined,
  "requestType": "read",
}
`;

exports[`regular getPlaces 2`] = `
Object {
  "method": "GET",
  "path": "/1/indexes/places/some_item",
  "qs": Object {
    "attributes": "one,name",
  },
  "requestOptions": undefined,
  "requestType": "read",
}
`;
